app/controllers/api/V1/vehicles_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Api::V1::VehiclesController < ApplicationController
+  # before_action :authenticate_admin!, only: %i[create destroy]
+
+  def create
+    @new_vehicle = Vehicle.new(api_params)
+
+    if @new_vehicle.save
+      render json: { message: 'Vehilcle saved successfully' }, status: :created
+    else
+      render json: { errors: @new_vehicle.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    @vehicle = Vehicle.find(params[:id])
+    render json: @vehicle, status: :ok
+  end
+
+  def index
+    @vehicles = Vehicle.order(created_at: :desc).all
+
+    if @vehicles.empty?
+      render json: { error: 'No vehicles found' }, status: :not_found
+    else
+      render json: @vehicles
+    end
+  end
+
+  def destroy
+    @vehicle = Vehicle.find(params[:id])
+    @vehicle.reservations.destroy_all
+    @vehicle.destroy
+    render json: { message: 'Vehicle deleted' }, status: :ok
+  end
```

Comment:
- [ ] What happens here if the vehicle is not successfully destroyed?  Kindly, add some error handling like you did in the index method (line 19). 👍